code/game/objects/structures/crates_lockers/closets/bodybag.dm

```diff
@@ -147,3 +148,163 @@
 		max_weight_of_contents = A_is_item.w_class
 	B.w_class = max_weight_of_contents
 	usr.put_in_hands(B)
+
+/// Environmental bags
+
+/obj/structure/closet/body_bag/environmental
+	name = "environmental protection bag"
+	desc = "An insulated, reinforced bag designed to protect against exoplanetary storms and other environmental factors."
+	icon = 'icons/obj/bodybag.dmi'
+	icon_state = "envirobag"
+	mob_storage_capacity = 1
+	pressure_protection = 0.8
+	thermal_insulation = 0.5
+	foldedbag_path = /obj/item/body_bag/environmental/
+	weather_protection = list("acid", "ash", "rad", "snow", "void") // Does not protect against lava or the The Floor Is Lava spell.
+
+/obj/structure/closet/body_bag/environmental/nanotrasen
+	name = "elite environmental protection bag"
+	desc = "Used for VIP extraction from hazardous environments and as shelter by elite Nanotrasen operatives, this highly reinforced bag is nearly completely isolated from environmental factors when closed."
+	icon = 'icons/obj/bodybag.dmi'
+	icon_state = "ntenvirobag"
+	pressure_protection = 1
+	thermal_insulation = 1
+	foldedbag_path = /obj/item/body_bag/environmental/nanotrasen/
+	weather_protection = list("all")
```

Comment:
i wish weather protection was a bitflag, so we didn't have to use a list and this could just be `ALL`

Reply:
I wish so too, but I'm structuring it as a list the same way that mob immunities are in a list.